### PR TITLE
[SR-8272] Drop the last remnants of LogicValue

### PR DESCRIPTION
**Pull request by CodaFi**
Removes the _getBuiltinLogicValue intrinsic in favor of an open-coded struct_extract in SIL.  This removes Sema's last non-literal use of builtin integer types and unblocks a bunch of cleanup.

This patch would be NFC, but it improves line information for conditional expression codegen.

Resolves [SR-8272](https://bugs.swift.org/browse/SR-8272).

Since this seems to have hit the trifecta of ABI change, compiler change, and stdlib change: lots of reviews please.  This oughta go into 5.0 as well.

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please test source compatibility

**Comment by slavapestov** [comment]
I don’t think this is an ABI change; you’re only removing a transparent function so it should have been inlined away. It’s up to you if you want to push for this to go into 5.0 or not, but my gut feeling is to be conservative and leave it on master.

@Azoy weren’t you working on something similar recently?

**Comment by Azoy** [comment]
Yeah I had something similar here: https://github.com/apple/swift/pull/21285, but this patch seems successful in removing this logic from Sema and into SILGen.

**Comment by Azoy** [comment]
While I don't think this affects ABI (because transparent), I think it is still a good candidate for the 5.0 branch because I don't think we really want users to be able to access this function as an API.
```swift
let x = true._getBuiltinLogicValue() // ok
```
The same could be said about the `_getBool` function I removed not to long ago as well.
```swift
let x = true._getBuiltinLogicValue() // ok
let y = _getBool(x) // ok
```

**Comment by CodaFi** [comment]
Source Compatibility passed (https://ci.swift.org/job/swift-PR-source-compat-suite/2331/ and https://ci.swift.org/job/swift-PR-source-compat-suite-debug/557/).  Fixing the API digester tests.

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by jrose-apple** [comment]
*Technically* transparent functions are only inlined away when called immediately, but this particular function is almost certainly never not called immediately.

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
Still want core team member sign-off on this and plans to push these to 5.0.

@DougGregor, @airspeedswift?

**Review by DougGregor — APPROVED** [review]
This is a very nice improvement. I don't think it change the ABI in practical terms because of the `@transparent`, but I would like to get it into 5.0 if we can. Let's get it merged to master and cherry-pick for 5.0.

**Comment by CodaFi** [comment]
⛵️

**Comment by DougGregor** [comment]
@CodaFi will you cherry-pick for the 5.0 branch?

**Comment by CodaFi** [comment]
Cherry picking as I type.

**Pull request merged**
